src/v0/destinations/braze/util.js

```diff
@@ -582,7 +582,7 @@ function getPurchaseObjs(message, config) {
     }
 
     products.forEach((product) => {
-      const productId = product.product_id || product.sku;
+      const productId = product?.product_id || product?.sku;
       const { price, quantity, currency: prodCurrencyCode } = product;
```

Comment:
It can fail for this destructuring as well if product is undefined. Better to first keep a if condition that product is defined in the loop

Reply:
Addressed